dtfs-central-api/test-helpers/get-fee-record-correction-url.ts

```diff
@@ -0,0 +1,22 @@
+type FeeRecordCorrectionPathParameters = {
+  reportId?: number | string;
+  feeRecordId?: number | string;
+  userId?: string;
+};
+
+/**
+ * Constructs a URL for fee record correction by replacing path parameters in
+ * the base URL.
+ * @param baseUrl - The base URL containing path parameter placeholders
+ * @param pathParameters - Object containing values for the path parameters
+ * @returns The constructed URL with path parameters replaced
+ * @throws Error if any required path parameter is missing or undefined
+ */
+export const getFeeRecordCorrectionUrl = (baseUrl: string, pathParameters: FeeRecordCorrectionPathParameters) =>
```

Comment:
I think `userId` isn't used in one of the routes so that should've been the only optional one. Fair point on 'reversing' the check to use the URL placeholders instead, I'll make that change thanks

Reply:
For the second comment, I was debating this too, and typing as something like `Record<string, string | number>`

I don't really want this ticket to cover replacing all the getUrl functions in these API tests though, so might pull that work out to another PR/ticket

Reply:
Yeah I think that is fine, this is enough for now and we can just do it as we go and touch things, raising a ticket isn't a bad idea in case there is specific time for tech debt to target using this in more places